tests/greendns_test.py

```diff
@@ -918,6 +923,57 @@ def test_hosts_priority():
     assert rrs6[0].address == 'dead:beef::1', rrs6[0].address
 
 
+def test_hosts_no_network():
+
+    name = 'example.com'
+    addr_from_ns = '1.0.2.0'
+    addr6_from_ns = 'dead:beef::1'
+
+    hr = _make_host_resolver()
+    rp = greendns.ResolverProxy(hosts_resolver=hr, filename=None)
+    base = _make_mock_base_resolver()
+    base.rr.address = addr_from_ns
+    base.rr6.address = addr6_from_ns
+    rp._resolver = base()
+    greendns.resolver = rp
```

Comment:
This modifies global state for all other tests. Please use `@/with mock.patch` or isolated test or `try/finally`.